spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/DatastoreEntityConverter.java

```diff
@@ -42,15 +42,16 @@
 
 	/**
 	 * Read the entity as a {@link Map}.
+	 * @param <T> the type of the key in the map
+	 * @param <R> the type of the value in the map
 	 * @param keyType the key type of the map to be read.
 	 * @param componentType the value type of the map, into which each field value will be
 	 * converted.
 	 * @param entity the entity from Cloud Datastore.
-	 * @param <T> the type of the key in the map
-	 * @param <R> the type of the value in the map
+	 * @param typeInformation the actual type of the map
 	 * @return a Map where the key values are the field names and the values the field
 	 * values.
 	 */
 	<T, R> Map<T, R> readAsMap(Class<T> keyType, TypeInformation<R> componentType,
-			BaseEntity entity);
+			BaseEntity entity, TypeInformation typeInformation);
```

Comment:
1. Technically, it's a breaking change. Can we just overload with a new method that has an additional argument?
This especially seems to make sense because you sometimes call the method with `null` `typeInformation`.
2. Also, can we rename `typeInformation` to `mapTypeInformation`?
3. If you're getting the type information for the whole map, why do you need the key and value type information? Maybe it would be easier if the caller just passed the map they wanted to fill with results?